### PR TITLE
Add hooks in packetio.Wrapper for custom messages (jacdac)

### PR DESCRIPTION
**Pull request by mmoskal**
@pelikhan you can add some routing between sim and wrapper if you want

### CODE REVIEW

**Comment by mmoskal** [comment]
@pelikhan this is now working in Maker; I'm not sure if I need any origin checks.

Also, it would be now good to stop Maker spinning two simulators on jacdac.

File: pxtlib/hf2.ts [hunk 1]
```diff
@@ -234,6 +239,9 @@ namespace pxt.HF2 {
                 }
                 //this.msgs.pushError(err)
             }
+            this.onEvent(HF2_EV_JDS_PACKET, buf => {
+                this.onCustomEvent("jacdac", buf)
```
Comment (pelikhan) [inline pxtlib/hf2.ts:243]: move "jacdac" to constant as well?

File: pxtlib/hf2.ts [hunk 2]
```diff
@@ -270,6 +280,13 @@ namespace pxt.HF2 {
             this.eventHandlers[id + ""] = f
         }
 
+        sendCustomEventAsync(type: string, payload: Uint8Array): Promise<void> {
+            if (type == "jacdac")
+                return this.talkAsync(HF2_CMD_JDS_SEND, payload)
+                    .then(() => { })
```
Comment (pelikhan) [inline pxtlib/hf2.ts:286]: what happens when we fail, just swallow the error?

**Review by pelikhan — APPROVED** [review]

**Comment by pelikhan** [comment]
> Also, it would be now good to stop Maker spinning two simulators on jacdac.

I've disabled it.

File: pxtlib/hf2.ts [hunk 3]
```diff
@@ -270,6 +282,16 @@ namespace pxt.HF2 {
             this.eventHandlers[id + ""] = f
         }
 
+        sendCustomEventAsync(type: string, payload: Uint8Array): Promise<void> {
+            if (type == CUSTOM_EV_JACDAC)
+                if (this.jacdacAvailable)
+                    return this.talkAsync(HF2_CMD_JDS_SEND, payload)
+                        .then(() => { })
```
Comment (pelikhan) [inline pxtlib/hf2.ts:289]: we should swallow the error here

File: webapp/src/app.tsx [hunk 4]
```diff
@@ -4096,7 +4096,27 @@ function initPacketIO() {
...
             }, "*")
+        },
+        (type, payload) => {
+            if (type == pxt.HF2.CUSTOM_EV_JACDAC) {
+                window.postMessage({
+                    type: "messagepacket",
+                    broadcast: false,
+                    channel: type,
+                    data: payload,
+                    fromPacketIO: true
```
Comment (pelikhan) [inline webapp/src/app.tsx:4107]: ``sender: "packetio"`` ? i'll handle that on the jacdac side

Comment (pelikhan) [reply webapp/src/app.tsx:4107]: if you merge origin/master, the ``SimulatorControlMessage`` is now in pxt.

File: webapp/src/app.tsx [hunk 5]
```diff
@@ -4096,7 +4096,27 @@ function initPacketIO() {
                 id: 'n/a', // TODO
                 data
             }, "*")
+        },
+        (type, payload) => {
+            if (type == pxt.HF2.CUSTOM_EV_JACDAC) {
```
Comment (pelikhan) [inline webapp/src/app.tsx:4101]: if the message name is in pxt.appTarget.simulator.messageSimulators, we could send it over, so we don't hardcode jacdac.

File: webapp/src/app.tsx [hunk 6]
```diff
@@ -4096,7 +4096,27 @@ function initPacketIO() {
...
+                    channel: type,
+                    data: payload,
+                    sender: "packetio",
+                }, "*")
+            }
         });
+
+    window.addEventListener('message', (ev: MessageEvent) => {
+        const msg = ev.data
+        if (msg.type == 'messagepacket' && msg.channel == pxt.HF2.CUSTOM_EV_JACDAC && msg.sender != "packetio")
```
Comment (pelikhan) [inline webapp/src/app.tsx:4114]: same thing here, look up channel in ``pxt.appTarget.simulator.messageSimulators``

**Comment by pelikhan** [comment]
I've hookup the events to the message simulator info in pxtarget.

**Pull request merged**